src/Raven.Studio/typescript/viewmodels/database/indexes/additionalSourceSyntax.ts

```diff
@@ -43,18 +61,39 @@ map('People', function (person) { return {
     };
 })`;
 
-    additionalSourceCsharpHtml = ko.pureComputed(() => {
-        return Prism.highlight(additionalSourceSyntax.additionalSourceCsharp, (Prism.languages as any).csharp);
-    });
-    usageInMapCsharpHtml = ko.pureComputed(() => {
-        return Prism.highlight(additionalSourceSyntax.usageInMapCsharp, (Prism.languages as any).csharp);
-    });
-    additionalSourceJavascriptHtml = ko.pureComputed(() => {
-        return Prism.highlight(additionalSourceSyntax.additionalSourceJavascript, (Prism.languages as any).javascript);
-    });
-    usageInMapJavascriptHtml = ko.pureComputed(() => {
-        return Prism.highlight(additionalSourceSyntax.usageInMapJavascript, (Prism.languages as any).javascript);
-    });
+    static readonly csharpSamples: Array<sampleCode> = [
+        {
+            title: "Csharp - Additional Source",
+            text: additionalSourceSyntax.additionalSourceCsharpText,
+            html: ko.pureComputed(() => {
+                return Prism.highlight(additionalSourceSyntax.additionalSourceCsharpText, (Prism.languages as any).csharp);
+            })
+        },
+        {
+            title: "Csharp - Usage in Map",
+            text: additionalSourceSyntax.usageInMapCsharpText,
+            html: ko.pureComputed(() => {
+                return Prism.highlight(additionalSourceSyntax.usageInMapCsharpText, (Prism.languages as any).csharp);
+            })
+        }
+    ];
     
+    static readonly javascriptSamples: Array<sampleCode> = [
+        {
+            title: "Javascript - Additional Source",
+            text: additionalSourceSyntax.additionalSourceJavascriptText,
+            html: ko.pureComputed(() => {
```

Comment:
do we need computed here? Looks like it can be computed one time.

Reply:
done